### PR TITLE
Add python tests for regorus

### PR DESCRIPTION
**Pull request by mgunnala**
This PR sets up some basic tests for the Regorus Python binding, using pytest.

### CODE REVIEW

**Review by yangjie-msft — APPROVED** [review]

File: bindings/python/test_extensions.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        }
+    }
+    data_json = json.dumps(data_json)
+    engine.add_data_json(data_json)
+    engine.set_input_json(input_data)
+    # Eval query
+    results = engine.eval_query('data.agent_extension_policy')
+    assert results['result'][0]['expressions'][0]['value']['extensions_to_download'][TEST_EXT_NAME]['downloadAllowed']
+
+
```
Comment (yangjie-msft) [inline bindings/python/test_extensions.py:103]: can you also add a case where the rule is extensionSigned False, and input has both extensionSigned true and false respectively and both are validated?

File: bindings/python/test_extensions.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+def test_default_data_file(engine, input_data):
+    """Test the default data in file format for extension policy."""
+    data_default_path = "../../examples/extension_list/agent-extension-default-data.json"
+    engine.add_data_from_json_file(data_default_path)
+    engine.set_input_json(input_data)
+    # Eval query
+    results = engine.eval_query('data.agent_extension_policy')
+    assert results['result'][0]['expressions'][0]['value']['extensions_to_download'][TEST_EXT_NAME]['downloadAllowed']
+
```
Comment (yangjie-msft) [inline bindings/python/test_extensions.py:83]: can you add a case that allowListOnly is true and download not allowed with your input, and a case where data includes the extension with allowListOnly to true, the download is allowed?

**Review by yangjie-msft — CHANGES_REQUESTED** [review]

File: bindings/python/test_extensions.py (hunk 2 above)
Comment (mgunnala) [reply bindings/python/test_extensions.py:83]: Added

File: bindings/python/test_extensions.py (hunk 1 above)
Comment (mgunnala) [reply bindings/python/test_extensions.py:103]: Added

**Review by yangjie-msft — APPROVED** [review]

**Pull request merged**